### DISCUSSION TITLE
应该整一个Xdi8-to-pinyin

### DISCUSSION THREAD

**Discussion by Honoka55** [edward-martyr/Xdi8Translator#12] [Ideas]
希顶→拼音或者希顶→希北，方便使用X Studio整活儿。（例活儿：[希顶国际歌](https://b23.tv/5TeblB) ）

不知道怎么pr（逃

**Comment by yuanhao-chen-nyoeghau** [comment]
Fork 以后 就可以提交 Pull request啦

**Comment by Kosmas131** [comment]
xdi82pinyin_dict={
    'ba': 'ba',
    'bo': 'bo',
    'be': 'be',
    'bE': 'bi',
    'bA': 'bu',
    'bY': 'bai',
    'bL': 'bao',
    'b6': 'bou',
    'b2': 'ban',
    'bT': 'bang',
    'b8': 'ben',
    'b3': 'bong',
    'bV': 'bei',
    'b1': 'bï',
    'bia': 'bia',
    'bio': 'bio',
    'bie': 'bie',
    'biA': 'bü',
    'biY': 'biai',
    'biL': 'biao',
    'bi6': 'biu',
    'bi2': 'bian',
    'biT': 'biang',
    'bi8': 'bin',
    'bi3': 'biong',
    'biV': 'biei',
    'pa': 'pa',
    'po': 'po',
    'pe': 'pe',
    'pE': 'pi',
    'pA': 'pu',
    'pY': 'pai',
    'pL': 'pao',
    'p6': 'pou',
    'p2': 'pan',
    'pT': 'pang',
    'p8': 'pen',
    'p3': 'pong',
    'pV': 'pei',
    'p1': 'pï',
    'pia': 'pia',
    'pio': 'pio',
    'pie': 'pie',
    'piA': 'pü',
    'piY': 'piai',
    'piL': 'piao',
    'pi6': 'piu',
    'pi2': 'pian',
    'piT': 'piang',
    'pi8': 'pin',
    'pi3': 'piong',
    'piV': 'piei',
    'ma': 'ma',
    'mo': 'mo',
    'me': 'me',
    'mE': 'mi',
    'mA': 'mu',
    'mY': 'mai',
    'mL': 'mao',
    'm6': 'mou',
    'm2': 'man',
    'mT': 'mang',
    'm8': 'men',
    'm3': 'mong',
    'mV': 'mei',
    'm1': 'mï',
    'mia': 'mia',
    'mio': 'mio',
    'mie': 'mie',
    'miA': 'mü',
    'miY': 'miai',
    'miL': 'miao',
    'mi6': 'miu',
    'mi2': 'mian',
    'miT': 'miang',
    'mi8': 'min',
    'mi3': 'miong',
    'miV': 'miei',
    'wa': 'wa',
    'wo': 'wo',
    'we': 'we',
    'wE': 'wi',
    'wA': 'wu',
    'wY': 'wai',
    'wL': 'wao',
    'w6': 'wou',
    'w2': 'wan',
    'wT': 'wang',
    'w8': 'wen',
    'w3': 'wong',
    'wV': 'wei',
    'w1': 'wï',
    'wia': 'wia',
    'wio': 'wio',
    'wie': 'wie',
    'wiA': 'wü',
    'wiY': 'wiai',
    'wiL': 'wiao',
    'wi6': 'wiu',
    'wi2': 'wian',
    'wiT': 'wiang',
    'wi8': 'win',
    'wi3': 'wiong',
    'wiV': 'wiei',
    'ja': 'jia',
    'jo': 'jio',
    'je': 'jie',
    'jE': 'ji',
    'jA': 'ju',
    'jY': 'jiai',
    'jL': 'jiao',
    'j6': 'jiu',
    'j2': 'jian',
    'jT': 'jiang',
    'j8': 'jin',
    'j3': 'jiong',
    'jV': 'jiei',
    'jua': 'jua',
    'juo': 'juo',
    'jue': 'jue',
    'juY': 'juai',
    'juL': 'juao',
    'ju6': 'juou',
    'ju2': 'juan',
    'juT': 'juang',
    'ju8': 'jun',
    'ju3': 'juong',
    'juV': 'jui',
    'qa': 'qia',
    'qo': 'qio',
    'qe': 'qie',
    'qE': 'qi',
    'qA': 'qu',
    'qY': 'qiai',
    'qL': 'qiao',
    'q6': 'qiu',
    'q2': 'qian',
    'qT': 'qiang',
    'q8': 'qin',
    'q3': 'qiong',
    'qV': 'qiei',
    'qua': 'qua',
    'quo': 'quo',
    'que': 'que',
    'quY': 'quai',
    'quL': 'quao',
    'qu6': 'quou',
    'qu2': 'quan',
    'quT': 'quang',
    'qu8': 'qun',
    'qu3': 'quong',
    'quV': 'qui',
    'xa': 'xia',
    'xo': 'xio',
    'xe': 'xie',
    'xE': 'xi',
    'xA': 'xu',
    'xY': 'xiai',
    'xL': 'xiao',
    'x6': 'xiu',
    'x2': 'xian',
    'xT': 'xiang',
    'x8': 'xin',
    'x3': 'xiong',
    'xV': 'xiei',
    'xua': 'xua',
    'xuo': 'xuo',
    'xue': 'xue',
    'xuY': 'xuai',
    'xuL': 'xuao',
    'xu6': 'xuou',
    'xu2': 'xuan',
    'xuT': 'xuang',
    'xu8': 'xun',
    'xu3': 'xuong',
    'xuV': 'xui',
    'ya': 'ya',
    'yo': 'yo',
    'ye': 'ye',
    'yE': 'yi',
    'yA': 'yu',
    'yY': 'yai',
    'yL': 'yao',
    'y6': 'you',
    'y2': 'yan',
    'yT': 'yang',
    'y8': 'yin',
    'y3': 'yong',
    'yV': 'yei',
    'yua': 'yua',
    'yuo': 'yuo',
    'yue': 'yue',
    'yuY': 'yuai',
    'yuL': 'yuao',
    'yu6': 'yuou',
    'yu2': 'yuan',
    'yuT': 'yuang',
    'yu8': 'yun',
    'yu3': 'yuong',
    'yuV': 'yui',
    'na': 'nia',
    'no': 'nio',
    'ne': 'nie',
    'nE': 'ni',
    'nA': 'nü',
    'nY': 'niai',
    'nL': 'niao',
    'n6': 'niu',
    'n2': 'nian',
    'nT': 'niang',
    'n8': 'nin',
    'n3': 'niong',
    'nV': 'niei',
    'nua': 'nüa',
    'nuo': 'nüo',
    'nue': 'nüe',
    'nuY': 'nüai',
    'nuL': 'nüao',
    'nu6': 'nüou',
    'nu2': 'nüan',
    'nuT': 'nüang',
    'nu8': 'nün',
    'nu3': 'nüong',
    'nuV': 'nüei',
    'za': 'za',
    'zo': 'zo',
    'ze': 'ze',
    'zE': 'zi',
    'zA': 'zu',
    'zY': 'zai',
    'zL': 'zao',
    'z6': 'zou',
    'z2': 'zan',
    'zT': 'zang',
    'z8': 'zen',
    'z3': 'zong',
    'zV': 'zei',
    'zua': 'zua',
    'zuo': 'zuo',
    'zue': 'zue',
    'zuY': 'zuai',
    'zuL': 'zuao',
    'zu6': 'zuou',
    'zu2': 'zuan',
    'zuT': 'zuang',
    'zu8': 'zun',
    'zu3': 'zuong',
    'zuV': 'zui',
    'Da': 'ca',
    'Do': 'co',
    'De': 'ce',
    'DE': 'ci',
    'DA': 'cu',
    'DY': 'cai',
    'DL': 'cao',
    'D6': 'cou',
    'D2': 'can',
    'DT': 'cang',
    'D8': 'cen',
    'D3': 'cong',
    'DV': 'cei',
    'Dua': 'cua',
    'Duo': 'cuo',
    'Due': 'cue',
    'DuY': 'cuai',
    'DuL': 'cuao',
    'Du6': 'cuou',
    'Du2': 'cuan',
    'DuT': 'cuang',
    'Du8': 'cun',
    'Du3': 'cuong',
    'DuV': 'cui',
    'sa': 'sa',
    'so': 'so',
    'se': 'se',
    'sE': 'si',
    'sA': 'su',
    'sY': 'sai',
    'sL': 'sao',
    's6': 'sou',
    's2': 'san',
    'sT': 'sang',
    's8': 'sen',
    's3': 'song',
    'sV': 'sei',
    'sua': 'sua',
    'suo': 'suo',
    'sue': 'sue',
    'suY': 'suai',
    'suL': 'suao',
    'su6': 'suou',
    'su2': 'suan',
    'suT': 'suang',
    'su8': 'sun',
    'su3': 'suong',
    'suV': 'sui',
    'ra': 'ra',
    'ro': 'ro',
    're': 're',
    'rE': 'ri',
    'rA': 'ru',
    'rY': 'rai',
    'rL': 'rao',
    'r6': 'rou',
    'r2': 'ran',
    'rT': 'rang',
    'r8': 'ren',
    'r3': 'rong',
    'rV': 'rei',
    'rua': 'rua',
    'ruo': 'ruo',
    'rue': 'rue',
    'ruY': 'ruai',
    'ruL': 'ruao',
    'ru6': 'ruou',
    'ru2': 'ruan',
    'ruT': 'ruang',
    'ru8': 'run',
    'ru3': 'ruong',
    'ruV': 'rui',
    'Ha': 'na',
    'Ho': 'no',
    'He': 'ne',
    'HE': 'nï',
    'HA': 'nu',
    'HY': 'nai',
    'HL': 'nao',
    'H6': 'nou',
    'H2': 'nan',
    'HT': 'nang',
    'H8': 'nen',
    'H3': 'nong',
    'HV': 'nei',
    'Hua': 'nua',
    'Huo': 'nuo',
    'Hue': 'nue',
    'HuY': 'nuai',
    'HuL': 'nuao',
    'Hu6': 'nuou',
    'Hu2': 'nuan',
    'HuT': 'nuang',
    'Hu8': 'nun',
    'Hu3': 'nuong',
    'HuV': 'nui',
    'Na': 'nga',
    'No': 'ngo',
    'Ne': 'nge',
    'NE': 'ngi',
    'NA': 'ngu',
    'NY': 'ngai',
    'NL': 'ngao',
    'N6': 'ngou',
    'N2': 'ngan',
    'NT': 'ngang',
    'N8': 'ngen',
    'N3': 'ngong',
    'NV': 'ngei',
    'N1': 'ngï',
    'Nia': 'ngia',
    'Nio': 'ngio',
    'Nie': 'ngie',
    'NiA': 'ngü',
    'NiY': 'ngiai',
    'NiL': 'ngiao',
    'Ni6': 'ngiu',
    'Ni2': 'ngian',
    'NiT': 'ngiang',
    'Ni8': 'ngin',
    'Ni3': 'ngiong',
    'NiV': 'ngiei',
    'la': 'la',
    'lo': 'lo',
    'le': 'le',
    'lE': 'li',
    'lA': 'lu',
    'lY': 'lai',
    'lL': 'lao',
    'l6': 'lou',
    'l2': 'lan',
    'lT': 'lang',
    'l8': 'len',
    'l3': 'long',
    'lV': 'lei',
    'l1': 'lï',
    'lia': 'lia',
    'lio': 'lio',
    'lie': 'lie',
    'liA': 'lü',
    'liY': 'liai',
    'liL': 'liao',
    'li6': 'liu',
    'li2': 'lian',
    'liT': 'liang',
    'li8': 'lin',
    'li3': 'liong',
    'liV': 'liei',
    'da': 'da',
    'do': 'do',
    'de': 'de',
    'dE': 'di',
    'dA': 'du',
    'dY': 'dai',
    'dL': 'dao',
    'd6': 'dou',
    'd2': 'dan',
    'dT': 'dang',
    'd8': 'den',
    'd3': 'dong',
    'dV': 'dei',
    'd1': 'dï',
    'dia': 'dia',
    'dio': 'dio',
    'die': 'die',
    'diA': 'dü',
    'diY': 'diai',
    'diL': 'diao',
    'di6': 'diu',
    'di2': 'dian',
    'diT': 'diang',
    'di8': 'din',
    'di3': 'diong',
    'diV': 'diei',
    'ta': 'ta',
    'to': 'to',
    'te': 'te',
    'tE': 'ti',
    'tA': 'tu',
    'tY': 'tai',
    'tL': 'tao',
    't6': 'tou',
    't2': 'tan',
    'tT': 'tang',
    't8': 'ten',
    't3': 'tong',
    'tV': 'tei',
    't1': 'tï',
    'tia': 'tia',
    'tio': 'tio',
    'tie': 'tie',
    'tiA': 'tü',
    'tiY': 'tiai',
    'tiL': 'tiao',
    'ti6': 'tiu',
    'ti2': 'tian',
    'tiT': 'tiang',
    'ti8': 'tin',
    'ti3': 'tiong',
    'tiV': 'tiei',
    'ga': 'ga',
    'go': 'go',
    'ge': 'ge',
    'gE': 'gi',
    'gA': 'gu',
    'gY': 'gai',
    'gL': 'gao',
    'g6': 'gou',
    'g2': 'gan',
    'gT': 'gang',
    'g8': 'gen',
    'g3': 'gong',
    'gV': 'gei',
    'g1': 'gï',
    'gia': 'gia',
    'gio': 'gio',
    'gie': 'gie',
    'giA': 'gü',
    'giY': 'giai',
    'giL': 'giao',
    'gi6': 'giu',
    'gi2': 'gian',
    'giT': 'giang',
    'gi8': 'gin',
    'gi3': 'giong',
    'giV': 'giei',
    'ka': 'ka',
    'ko': 'ko',
    'ke': 'ke',
    'kE': 'ki',
    'kA': 'ku',
    'kY': 'kai',
    'kL': 'kao',
    'k6': 'kou',
    'k2': 'kan',
    'kT': 'kang',
    'k8': 'ken',
    'k3': 'kong',
    'kV': 'kei',
    'k1': 'kï',
    'kia': 'kia',
    'kio': 'kio',
    'kie': 'kie',
    'kiA': 'kü',
    'kiY': 'kiai',
    'kiL': 'kiao',
    'ki6': 'kiu',
    'ki2': 'kian',
    'kiT': 'kiang',
    'ki8': 'kin',
    'ki3': 'kiong',
    'kiV': 'kiei',
    'ha': 'ha',
    'ho': 'ho',
    'he': 'he',
    'hE': 'hi',
    'hA': 'hu',
    'hY': 'hai',
    'hL': 'hao',
    'h6': 'hou',
    'h2': 'han',
    'hT': 'hang',
    'h8': 'hen',
    'h3': 'hong',
    'hV': 'hei',
    'h1': 'hï',
    'hia': 'hia',
    'hio': 'hio',
    'hie': 'hie',
    'hiA': 'hü',
    'hiY': 'hiai',
    'hiL': 'hiao',
    'hi6': 'hiu',
    'hi2': 'hian',
    'hiT': 'hiang',
    'hi8': 'hin',
    'hi3': 'hiong',
    'hiV': 'hiei',
    'hua': 'hüa',
    'huo': 'hüo',
    'hue': 'hüe',
    'huE': 'hüi',
    'huY': 'hüai',
    'huL': 'hüao',
    'hu6': 'hüou',
    'hu2': 'hüan',
    'huT': 'hüang',
    'hu8': 'hün',
    'hu3': 'hüong',
    'huV': 'hüei',
    '4a': 'a',
    '4o': 'o',
    '4e': 'e',
    '4E': 'j̈i',
    '4A': 'u',
    '4Y': 'ai',
    '4L': 'ao',
    '46': 'ou',
    '42': 'an',
    '4T': 'ang',
    '48': 'en',
    '43': 'ong',
    '4V': 'ei',
    '41': 'ï',
    '4ia': 'ia',
    '4io': 'io',
    '4ie': 'ie',
    '4iA': 'ü',
    '4iY': 'iai',
    '4iL': 'iao',
    '4i6': 'iu',
    '4i2': 'ian',
    '4iT': 'iang',
    '4i8': 'in',
    '4i3': 'iong',
    '4iV': 'iei',
    '4ua': 'üa',
    '4uo': 'üo',
    '4ue': 'üe',
    '4uE': 'üi',
    '4uY': 'üai',
    '4uL': 'üao',
    '4u6': 'üou',
    '4u2': 'üan',
    '4uT': 'üang',
    '4u8': 'ün',
    '4u3': 'üong',
    '4uV': 'üei',
    '5a': 'ngua',
    '5o': 'nguo',
    '5e': 'ngue',
    '5E': 'ngüi',
    '5Y': 'nguai',
    '5L': 'nguao',
    '56': 'nguou',
    '52': 'nguan',
    '5T': 'nguang',
    '58': 'ngun',
    '53': 'nguong',
    '5V': 'ngui',
    '5ia': 'ngüa',
    '5io': 'ngüo',
    '5ie': 'ngüe',
    '5iY': 'ngüai',
    '5iL': 'ngüao',
    '5i6': 'ngüou',
    '5i2': 'ngüan',
    '5iT': 'ngüang',
    '5i8': 'ngün',
    '5i3': 'ngüong',
    '5iV': 'ngüei',
    'va': 'lua',
    'vo': 'luo',
    've': 'lue',
    'vE': 'lüi',
    'vY': 'luai',
    'vL': 'luao',
    'v6': 'luou',
    'v2': 'luan',
    'vT': 'luang',
    'v8': 'lun',
    'v3': 'luong',
    'vV': 'lui',
    'via': 'lüa',
    'vio': 'lüo',
    'vie': 'lüe',
    'viY': 'lüai',
    'viL': 'lüao',
    'vi6': 'lüou',
    'vi2': 'lüan',
    'viT': 'lüang',
    'vi8': 'lün',
    'vi3': 'lüong',
    'viV': 'lüei',
    'Fa': 'dua',
    'Fo': 'duo',
    'Fe': 'due',
    'FE': 'düi',
    'FY': 'duai',
    'FL': 'duao',
    'F6': 'duou',
    'F2': 'duan',
    'FT': 'duang',
    'F8': 'dun',
    'F3': 'duong',
    'FV': 'dui',
    'Fia': 'düa',
    'Fio': 'düo',
    'Fie': 'düe',
    'FiY': 'düai',
    'FiL': 'düao',
    'Fi6': 'düou',
    'Fi2': 'düan',
    'FiT': 'düang',
    'Fi8': 'dün',
    'Fi3': 'düong',
    'FiV': 'düei',
    '7a': 'tua',
    '7o': 'tuo',
    '7e': 'tue',
    '7E': 'tüi',
    '7Y': 'tuai',
    '7L': 'tuao',
    '76': 'tuou',
    '72': 'tuan',
    '7T': 'tuang',
    '78': 'tun',
    '73': 'tuong',
    '7V': 'tui',
    '7ia': 'tüa',
    '7io': 'tüo',
    '7ie': 'tüe',
    '7iY': 'tüai',
    '7iL': 'tüao',
    '7i6': 'tüou',
    '7i2': 'tüan',
    '7iT': 'tüang',
    '7i8': 'tün',
    '7i3': 'tüong',
    '7iV': 'tüei',
    'Ba': 'gua',
    'Bo': 'guo',
    'Be': 'gue',
    'BE': 'güi',
    'BY': 'guai',
    'BL': 'guao',
    'B6': 'guou',
    'B2': 'guan',
    'BT': 'guang',
    'B8': 'gun',
    'B3': 'guong',
    'BV': 'gui',
    'Bia': 'güa',
    'Bio': 'güo',
    'Bie': 'güe',
    'BiY': 'güai',
    'BiL': 'güao',
    'Bi6': 'güou',
    'Bi2': 'güan',
    'BiT': 'güang',
    'Bi8': 'gün',
    'Bi3': 'güong',
    'BiV': 'güei',
    'ca': 'kua',
    'co': 'kuo',
    'ce': 'kue',
    'cE': 'küi',
    'cY': 'kuai',
    'cL': 'kuao',
    'c6': 'kuou',
    'c2': 'kuan',
    'cT': 'kuang',
    'c8': 'kun',
    'c3': 'kuong',
    'cV': 'kui',
    'cia': 'küa',
    'cio': 'küo',
    'cie': 'küe',
    'ciY': 'küai',
    'ciL': 'küao',
    'ci6': 'küou',
    'ci2': 'küan',
    'ciT': 'küang',
    'ci8': 'kün',
    'ci3': 'küong',
    'ciV': 'küei',
    'fa': 'fa',
    'fo': 'fo',
    'fe': 'fe',
    'fE': 'fi',
    'fA': 'fu',
    'fY': 'fai',
    'fL': 'fao',
    'f6': 'fou',
    'f2': 'fan',
    'fT': 'fang',
    'f8': 'fen',
    'f3': 'fong',
    'fV': 'fei',
    'f1': 'fï',
    'fia': 'fia',
    'fio': 'fio',
    'fie': 'fie',
    'fiA': 'fü',
    'fiY': 'fiai',
    'fiL': 'fiao',
    'fi6': 'fiu',
    'fi2': 'fian',
    'fiT': 'fiang',
    'fi8': 'fin',
    'fi3': 'fiong',
    'fiV': 'fiei',
    'bua': 'ba',
    'buo': 'bo',
    'bue': 'be',
    'buE': 'bi',
    'buY': 'bai',
    'buL': 'bao',
    'bu6': 'bou',
    'bu2': 'ban',
    'buT': 'bang',
    'bu8': 'ben',
    'bu3': 'bong',
    'buV': 'bei',
    'bu1': 'bï',
    'pua': 'pa',
    'puo': 'po',
    'pue': 'pe',
    'puE': 'pi',
    'puY': 'pai',
    'puL': 'pao',
    'pu6': 'pou',
    'pu2': 'pan',
    'puT': 'pang',
    'pu8': 'pen',
    'pu3': 'pong',
    'puV': 'pei',
    'pu1': 'pï',
    'mua': 'ma',
    'muo': 'mo',
    'mue': 'me',
    'muE': 'mi',
    'muY': 'mai',
    'muL': 'mao',
    'mu6': 'mou',
    'mu2': 'man',
    'muT': 'mang',
    'mu8': 'men',
    'mu3': 'mong',
    'muV': 'mei',
    'mu1': 'mï',
    'jia': 'jia',
    'jio': 'jio',
    'jie': 'jie',
    'jiA': 'ju',
    'jiY': 'jiai',
    'jiL': 'jiao',
    'ji6': 'jiu',
    'ji2': 'jian',
    'jiT': 'jiang',
    'ji8': 'jin',
    'ji3': 'jiong',
    'jiV': 'jiei',
    'qia': 'qia',
    'qio': 'qio',
    'qie': 'qie',
    'qiA': 'qu',
    'qiY': 'qiai',
    'qiL': 'qiao',
    'qi6': 'qiu',
    'qi2': 'qian',
    'qiT': 'qiang',
    'qi8': 'qin',
    'qi3': 'qiong',
    'qiV': 'qiei',
    'xia': 'xia',
    'xio': 'xio',
    'xie': 'xie',
    'xiA': 'xu',
    'xiY': 'xiai',
    'xiL': 'xiao',
    'xi6': 'xiu',
    'xi2': 'xian',
    'xiT': 'xiang',
    'xi8': 'xin',
    'xi3': 'xiong',
    'xiV': 'xiei',
    'yia': 'ya',
    'yio': 'yo',
    'yie': 'ye',
    'yiA': 'yu',
    'yiY': 'yai',
    'yiL': 'yao',
    'yi6': 'you',
    'yi2': 'yan',
    'yiT': 'yang',
    'yi8': 'yin',
    'yi3': 'yong',
    'yiV': 'yei',
    'nia': 'nia',
    'nio': 'nio',
    'nie': 'nie',
    'niA': 'nü',
    'niY': 'niai',
    'niL': 'niao',
    'ni6': 'niu',
    'ni2': 'nian',
    'niT': 'niang',
    'ni8': 'nin',
    'ni3': 'niong',
    'niV': 'niei',
    'zia': 'za',
    'zio': 'zo',
    'zie': 'ze',
    'ziA': 'zu',
    'ziY': 'zai',
    'ziL': 'zao',
    'zi6': 'zou',
    'zi2': 'zan',
    'ziT': 'zang',
    'zi8': 'zen',
    'zi3': 'zong',
    'ziV': 'zei',
    'Dia': 'ca',
    'Dio': 'co',
    'Die': 'ce',
    'DiA': 'cu',
    'DiY': 'cai',
    'DiL': 'cao',
    'Di6': 'cou',
    'Di2': 'can',
    'DiT': 'cang',
    'Di8': 'cen',
    'Di3': 'cong',
    'DiV': 'cei',
    'sia': 'sa',
    'sio': 'so',
    'sie': 'se',
    'siA': 'su',
    'siY': 'sai',
    'siL': 'sao',
    'si6': 'sou',
    'si2': 'san',
    'siT': 'sang',
    'si8': 'sen',
    'si3': 'song',
    'siV': 'sei',
    'ria': 'ra',
    'rio': 'ro',
    'rie': 're',
    'riA': 'ru',
    'riY': 'rai',
    'riL': 'rao',
    'ri6': 'rou',
    'ri2': 'ran',
    'riT': 'rang',
    'ri8': 'ren',
    'ri3': 'rong',
    'riV': 'rei',
    'Hia': 'na',
    'Hio': 'no',
    'Hie': 'ne',
    'HiA': 'nu',
    'HiY': 'nai',
    'HiL': 'nao',
    'Hi6': 'nou',
    'Hi2': 'nan',
    'HiT': 'nang',
    'Hi8': 'nen',
    'Hi3': 'nong',
    'HiV': 'nei',
    'Nua': 'ngua',
    'Nuo': 'nguo',
    'Nue': 'ngue',
    'NuE': 'ngüi',
    'NuY': 'nguai',
    'NuL': 'nguao',
    'Nu6': 'nguou',
    'Nu2': 'nguan',
    'NuT': 'nguang',
    'Nu8': 'ngun',
    'Nu3': 'nguong',
    'NuV': 'ngui',
    'lua': 'lua',
    'luo': 'luo',
    'lue': 'lue',
    'luE': 'lüi',
    'luY': 'luai',
    'luL': 'luao',
    'lu6': 'luou',
    'lu2': 'luan',
    'luT': 'luang',
    'lu8': 'lun',
    'lu3': 'luong',
    'luV': 'lui',
    'dua': 'dua',
    'duo': 'duo',
    'due': 'due',
    'duE': 'düi',
    'duY': 'duai',
    'duL': 'duao',
    'du6': 'duou',
    'du2': 'duan',
    'duT': 'duang',
    'du8': 'dun',
    'du3': 'duong',
    'duV': 'dui',
    'tua': 'tua',
    'tuo': 'tuo',
    'tue': 'tue',
    'tuE': 'tüi',
    'tuY': 'tuai',
    'tuL': 'tuao',
    'tu6': 'tuou',
    'tu2': 'tuan',
    'tuT': 'tuang',
    'tu8': 'tun',
    'tu3': 'tuong',
    'tuV': 'tui',
    'gua': 'gua',
    'guo': 'guo',
    'gue': 'gue',
    'guE': 'güi',
    'guY': 'guai',
    'guL': 'guao',
    'gu6': 'guou',
    'gu2': 'guan',
    'guT': 'guang',
    'gu8': 'gun',
    'gu3': 'guong',
    'guV': 'gui',
    'kua': 'kua',
    'kuo': 'kuo',
    'kue': 'kue',
    'kuE': 'küi',
    'kuY': 'kuai',
    'kuL': 'kuao',
    'ku6': 'kuou',
    'ku2': 'kuan',
    'kuT': 'kuang',
    'ku8': 'kun',
    'ku3': 'kuong',
    'kuV': 'kui',
    'bi': 'bi',
    'bu': 'bu',
    'pi': 'pi',
    'pu': 'pu',
    'mi': 'mi',
    'mu': 'mu',
    'wi': 'wi',
    'wu': 'wu',
    'ji': 'ji',
    'ju': 'ju',
    'qi': 'qi',
    'qu': 'qu',
    'xi': 'xi',
    'xu': 'xu',
    'yi': 'yi',
    'yu': 'yu',
    'ni': 'ni',
    'nu': 'nu',
    'zi': 'zi',
    'zu': 'zu',
    'Di': 'ci',
    'Du': 'cu',
    'si': 'si',
    'su': 'su',
    'ri': 'ri',
    'ru': 'ru',
    'Hi': 'nï',
    'Hu': 'nu',
    'Ni': 'ngi',
    'Nu': 'ngu',
    'li': 'li',
    'lu': 'lu',
    'di': 'di',
    'du': 'du',
    'ti': 'ti',
    'tu': 'tu',
    'gi': 'gi',
    'gu': 'gu',
    'ki': 'ki',
    'ku': 'ku',
    'hi': 'hi',
    'hu': 'hu',
    '4i': 'i',
    '4u': 'u',
    '5i': 'ngüi',
    'vi': 'lüi',
    'Fi': 'düi',
    '7i': 'tüi',
    'Bi': 'güi',
    'ci': 'küi',
    'fi': 'fi',
    'fu': 'fu',
    'b': 'bo',
    'p': 'po',
    'm': 'mo',
    'w': 'wo',
    'j': 'ji',
    'q': 'qi',
    'x': 'xi',
    'y': 'wai',
    'n': 'ni',
    'z': 'zi',
    'D': 'ci',
    's': 'si',
    'r': 'ri',
    'H': 'nï',
    'N': 'ngï',
    'l': 'lï',
    'd': 'de',
    't': 'te',
    'g': 'ge',
    'k': 'ke',
    'h': 'he',
    '4': 'qian',
    '5': 'ngu',
    'v': 'lu',
    'F': 'du',
    '7': 'tu',
    'B': 'gu',
    'c': 'ku',
    'f': 'fu',
    'u': 'wu',
    'a': 'a',
    'o': 'o',
    'e': 'e',
    'E': 'yi',
    'A': 'yu',
    'Y': 'ai',
    'L': 'ao',
    '6': 'ou',
    '2': 'an',
    'T': 'ang',
    '8': 'en',
    '3': 'ong',
    'V': 'ei',
    '1': 'ï',
    'i': 'i'
}
